src/DynamoUtilities/EnumerableExtensions.cs

```diff
@@ -467,5 +468,30 @@ where attribute.Name.Equals("value")
 
             return string.Empty;
         }
+
+        public static bool TryGetValueAs<Key, Value, ValueAs>(this IDictionary<Key, Value> dictionary, Key key, out ValueAs valueAs) where ValueAs : Value
+        {
+            if (dictionary.TryGetValue(key, out Value value))
+            {
+                valueAs = (ValueAs)value;
+                return true;
+            }
+
+            valueAs = default;
+            return false;
+        }
+
+
+        public static bool TryRemoveAs<Key, Value, ValueAs>(this ConcurrentDictionary<Key, Value> dictionary, Key key, out ValueAs valueAs) where ValueAs : Value
```

Comment:
Why is it that you have `ConcurrentDictionary` as the first parameter here while in the previous function you're using `IDictionary`?

Reply:
I will switch both to ConcurrentDictionary since that is the only place they are used at the moment